cluster/images/karmada-agent/Dockerfile

```diff
@@ -1,7 +1,20 @@
+FROM golang:1.17-alpine3.15 AS builder
+
+WORKDIR /go/src/github.com/karmada-io/karmada
+COPY . /go/src/github.com/karmada-io/karmada
+
+ARG GO_LDFLAGS
+
+RUN CGO_ENABLED=0 go build \
+    -ldflags "$GO_LDFLAGS" \
+    -o /bin/karmada-agent \
+    cmd/agent/main.go
+
```

Comment:
On my side, GOARCH must be set in build, or it still build as the same arch as building machine.

```
RUN CGO_ENABLED=0 GO111MODULE=off GOOS=${TARGETOS} GOARCH=${TARGETARCH} \
    go build ...
```

`TARGETOS` and `TARGETARCH` will be injected by buildx.

Reply:
Sorry, I didn't get what you mean.

Reply:
I guess you mean the normal endpoint, like `image-karmada-agent`, not the ~~mp-image-karmada-agent~~. Am I right?

Reply:
I reproduce this on my side. 
I tried to build `arm64` images on my `amd64` machine, but I got the `amd64` image:

```
-bash-4.2# export GOARCH=arm64
-bash-4.2# make image-karmada-agent
VERSION=v1.1.0-319-gaa39b35d REGISTRY="swr.ap-southeast-1.myhuaweicloud.com/karmada" hack/docker.sh karmada-agent
Step 5/9 : RUN CGO_ENABLED=0 go build     -ldflags "$GO_LDFLAGS"     -o /bin/karmada-agent     cmd/agent/main.go
...
Successfully tagged swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-agent:v1.1.0-319-gaa39b35d
```

Run the image and expect something like `WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested`, but I got:
```
-bash-4.2# docker run swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-agent:v1.1.0-319-gaa39b35d
Error: Options.ClusterName: Invalid value: "": must be not empty
Usage:
  karmada-agent [flags]

```

Reply:
`mp-image-xxx` i am saying.  When we run `docker buildx build --platform=linux/amd64,linux/arm64`, buildx will create two job: one with  arg `TARGETARCH=amd64`, and another `TARGETARCH=amr64`. But both `BUILDARCH` is amd64 or amr64, depending building macine.

So when we run `go build` in docker build, it still generate binary in the `BUILDARCH`. Another words, the two job create the same arch binary.

Unfortunately, my build env down, i cannot show you detail.

Reply:
but by test, docker buildx will build two different arch images, included binaries in images

https://github.com/karmada-io/karmada/pull/927#issuecomment-1110497417

Reply:
ok, I will test it again when my build env recover.

Reply:
You are right, binaries are different.

```
➜ file buildx/*/bin/karmada-aggregated-apiserver
buildx/linux_amd64/bin/karmada-aggregated-apiserver: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=DuIyRl7Z2PXkepVo1ahi/DsVwkQDyRqjh4uoczV8b/LavqXD9P1_jG2c90N0GA/lFZ71MwLdCi94dU6-2Tx, not stripped
buildx/linux_arm64/bin/karmada-aggregated-apiserver: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=dUbIM_EWa8BRqh8-Vy55/IIyI9BnL0EBhJJFhAfiV/IzNDZnca71YDBp6g3r4D/eKPDnCSeNsAnyBHD0F13, not stripped
```

Reply:
I know why i fail before. I addd `--platform=$BUILDPLATFORM` in dockerfile, its a mistake.

```
FROM --platform=$BUILDPLATFORM golang:1.17.8-alpine3.15 AS builder
```

Reply:
> Run the image and expect something like `WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested`, but I got:

first, docker build cannot build cross platform, so you get amd image on your amd machine.

then `GOARCH=arm64` is not delivered to docker build with `ARG`, finally `go build` not get this value， so it build binary with `amd64`.


By the way, is it necessary supporting cross build for `make image`?  `make image` most used for local test. We provide multi platform with `make mp-image`, it is enough.